eth/tracers/internal/tracers/call_tracer.js

```diff
@@ -14,212 +14,82 @@
-				if (call.gas !== undefined) {
-					call.gasUsed = '0x' + bigInt(call.gasIn - call.gasCost + call.gas - log.getGas()).toString(16);
-				}
-				var ret = log.stack.peek(0);
-				if (!ret.equals(0)) {
-					call.output = toHex(log.memory.slice(call.outOff, call.outOff + call.outLen));
-				} else if (call.error === undefined) {
-					call.error = "internal failure"; // TODO(karalabe): surface these faults somehow
-				}
-				delete call.gasIn; delete call.gasCost;
-				delete call.outOff; delete call.outLen;
-			}
-			if (call.gas !== undefined) {
-				call.gas = '0x' + bigInt(call.gas).toString(16);
-			}
-			// Inject the call into the previous one
-			var left = this.callstack.length;
-			if (this.callstack[left-1].calls === undefined) {
-				this.callstack[left-1].calls = [];
-			}
-			this.callstack[left-1].calls.push(call);
-		}
-	},
-
-	// fault is invoked when the actual execution of an opcode fails.
-	fault: function(log, db) {
-		// If the topmost call already reverted, don't handle the additional fault again
-		if (this.callstack[this.callstack.length - 1].error !== undefined) {
-			return;
-		}
-		// Pop off the just failed call
-		var call = this.callstack.pop();
-		call.error = log.getError();
-
-		// Consume all available gas and clean any leftovers
-		if (call.gas !== undefined) {
-			call.gas = '0x' + bigInt(call.gas).toString(16);
-			call.gasUsed = call.gas
-		}
-		delete call.gasIn; delete call.gasCost;
-		delete call.outOff; delete call.outLen;
-
-		// Flatten the failed call into its parent
-		var left = this.callstack.length;
-		if (left > 0) {
-			if (this.callstack[left-1].calls === undefined) {
-				this.callstack[left-1].calls = [];
-			}
-			this.callstack[left-1].calls.push(call);
-			return;
-		}
-		// Last call failed too, leave it in the stack
-		this.callstack.push(call);
-	},
-
-	// result is invoked when all the opcodes have been iterated over and returns
-	// the final result of the tracing.
-	result: function(ctx, db) {
-		var result = {
-			type:    ctx.type,
-			from:    toHex(ctx.from),
-			to:      toHex(ctx.to),
-			value:   '0x' + ctx.value.toString(16),
-			gas:     '0x' + bigInt(ctx.gas).toString(16),
-			gasUsed: '0x' + bigInt(ctx.gasUsed).toString(16),
-			input:   toHex(ctx.input),
-			output:  toHex(ctx.output),
-			time:    ctx.time,
-		};
-		if (this.callstack[0].calls !== undefined) {
-			result.calls = this.callstack[0].calls;
-		}
-		if (this.callstack[0].error !== undefined) {
-			result.error = this.callstack[0].error;
-		} else if (ctx.error !== undefined) {
-			result.error = ctx.error;
-		}
-		if (result.error !== undefined && (result.error !== "execution reverted" || result.output ==="0x")) {
-			delete result.output;
-		}
-		return this.finalize(result);
-	},
+// callFrameTracer uses the new call frame tracing methods to report useful information
+// about internal messages of a transaction.
+{
+    callstack: [{}],
+    step: function(log, db) {},
```

Comment:
Do you need to declare this function here? Wouldn't this defeat the purpose of not having to call back into js for every step?

Reply:
Oof good catch

Reply:
Ooof! I need to rerun the benches now!

Reply:
![image](https://user-images.githubusercontent.com/1591639/133597961-7026114b-fad9-47cb-b12c-61bb17fc6548.png)